packages/fxa-react/components/PasswordInput/index.tsx

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState, useCallback, ChangeEvent, FocusEvent } from 'react';
+import TextInput, { TextInputProps } from '../TextInput';
+import { ReactComponent as OpenEye } from './eye-open.svg';
+import { ReactComponent as ClosedEye } from './eye-closed.svg';
+
+type PasswordInputProps = Omit<TextInputProps, 'type'>;
+
+export const PasswordInput = ({
+  defaultValue,
+  disabled,
+  id,
+  label,
+  onBlur,
+  onChange,
+  onFocus,
+  onKeyPress,
+  placeholder,
+}: PasswordInputProps) => {
+  const [focussed, setFocussed] = useState<boolean>(false);
+  const [hasContent, setHasContent] = useState<boolean>(defaultValue != null);
+  const [visible, setVisible] = useState<boolean>(false);
+
+  const inputFocus = useCallback(
+    (event: FocusEvent<HTMLInputElement>) => {
+      setFocussed(true);
+      onFocus && onFocus(event);
+    },
+    [focussed]
+  );
+
+  const inputBlur = useCallback(
+    (event: FocusEvent<HTMLInputElement>) => {
+      setFocussed(false);
+      onBlur && onBlur(event);
+    },
+    [focussed]
+  );
+
+  const inputChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      setHasContent(event.target.value.length > 0);
+      onChange && onChange(event);
+    },
+    [hasContent]
+  );
+
+  return (
+    <TextInput
+      type={visible ? `text` : 'password'}
+      onFocus={inputFocus}
+      onBlur={inputBlur}
+      onChange={inputChange}
+      {...{
+        defaultValue,
+        disabled,
+        id,
+        label,
+        onKeyPress,
+        placeholder,
+      }}
+      extras={
```

Comment:
It could be appropriate to pass this in as `children` instead of an `extras` prop if it's always going to be React elements.